waiter/src/waiter/scheduler/marathon.clj

```diff
@@ -395,11 +419,143 @@
   (service-id->state [_ service-id]
     {:failed-instances (service-id->failed-instances service-id->failed-instances-transient-store service-id)
      :killed-instances (scheduler/service-id->killed-instances service-id)
-     :kill-info (get @service-id->kill-info-store service-id)})
+     :kill-info (get @service-id->kill-info-store service-id)
+     :out-of-sync-state (get @service-id->out-of-sync-state-store service-id)})
 
   (state [_]
     {:service-id->failed-instances-transient-store @service-id->failed-instances-transient-store
-     :service-id->kill-info-store @service-id->kill-info-store}))
+     :service-id->kill-info-store @service-id->kill-info-store
+     :service-id->out-of-sync-state-store @service-id->out-of-sync-state-store}))
+
+(defn- get-apps-with-deployments
+  "Retrieves the apps with the deployment info embedded."
+  [{:keys [marathon-api is-waiter-app?-fn]}]
+  (get-apps marathon-api is-waiter-app?-fn {"embed" ["apps.deployments" "apps.tasks"]}))
+
+(defn retrieve-service-id->out-of-sync-state
```

Comment:
I'd agree with Steven for this function. This function is only called in once place. I think it would make more sense to destructure `{:keys [service-id->out-of-sync-state-store] :as marathon-scheduler}` in that function, and replace the current `(retrieve-service-id->out-of-sync-state marathon-scheduler)` call with `@service-id->out-of-sync-state-store`. That would be more consistent with the `@service-id->out-of-sync-state-store` on line 428 above.